### PR TITLE
Fix both parts of #7924

### PR DESCRIPTION
**Pull request by aafemt**

### CODE REVIEW

File: src/dsql/ddl.cpp [hunk 1]
```diff
@@ -276,8 +276,9 @@ void DDL_resolve_intl_type(DsqlCompilerScratch* dsqlScratch, dsql_fld* field,
 		return;
 	}
 
-	if (modifying)
+	if (modifying && !field->charSet.hasData() && !field->collate.hasData())
```
Comment (asfernandes) [inline src/dsql/ddl.cpp:279]: `! .hasData()` -> `.isEmpty()`

Comment (aafemt) [reply src/dsql/ddl.cpp:279]: May be just `.empty()` then for STL compatibility?

File: src/jrd/MetaName.h [hunk 2]
```diff
@@ -258,6 +258,11 @@ class MetaName
 		return !word;
 	}
 
+	bool empty() const
+	{
+		return !word;
+	}
+
```
Comment (dyemanov) [inline src/jrd/MetaName.h:265]: I see no sense to have a duplication. If we're OK with STL-like `empty()`, then let's just rename `isEmpty()` across the codebase.

Comment (aafemt) [reply src/jrd/MetaName.h:265]: It would be too big and unrelated change for this PR but I can prepare a separate one if agreed.

**Comment by asfernandes** [comment]
What will happen if the field is used in an index and you change its collation?

**Comment by aafemt** [comment]
Most likely some kind of separate issue which should be fixed separately.

**Comment by asfernandes** [comment]
I don't think it is and actually I see more sense in not change the collation or throw error than have an index that will make queries return incorrect data.

**Comment by aafemt** [comment]
But as I said, it is a separate issue. This PR is for #7924 only.

**Review by dyemanov — APPROVED** [review]

**Pull request merged**